### DISCUSSION TITLE
Programatically set label to the current date

### DISCUSSION THREAD

**Discussion by lloop** [TriliumNext/Trilium#3428] [General]
I am trying to set the todoDate label in the TODO scritping code. This original issue seemed to be promising but it doesn't work. What I get is a date in quotes (obviously because it's a string).  #todoDate="12/17/2022" . But that doesn't show up in the note as a date. When I select a date from the calendar widget I get a date without quotes. So I believe it is creating a dayJS object. But when I use the object I get a date that is ISO format with T and the seconds. Doesn't work either. I am probably doing something stupid but can't figure it out.
TIA

### Discussed in https://github.com/zadam/trilium/discussions/2133

<div type='discussions-op-text'>

<sup>Originally posted by **jiyosub** August 29, 2021</sup>
Hi there.  I am extending the attribute changed script from the Task manager example.  I'd like to set the label doneDate to the current date when another attribute is changed.  Something like:

```
note.setLabel('doneDate', utils.localNowDateTime())
```

I can't figure out how to get the current datetime string. Any help?</div>

**Comment by zadam** [comment]
Hi, to be able to help you need to post some code samples.

**Reply by lloop** [reply]
const todayDate = api.dayjs();
const tommorrowDate = todayDate.add(1, 'day').format('MM/DD/YYYY');

doneNote.setLabel('todoDate', tommorrowDate);
doneNote.toggleLabel(false, 'doneDate');

**Reply by zadam** [reply]
Looks pretty good to me, but you need to use the ISO date format - `YYYY-MM-DD`.

**Comment by lloop** [comment]
Thanks!!!!!
